clarity_ext/service/file_service.py

```diff
@@ -318,6 +299,35 @@ def _local_shared_file(self, artifact, filename, mode='r', extension="",
         self.file_service._local_shared_files.append(f)
         return f
 
+    def _download_or_create_local_file(self, artifact, local_file_name_abs_path, modify_attached):
+        if not self.os_service.exists(local_file_name_abs_path) and len(artifact.files) == 0 and modify_attached:
+            # No file has been uploaded yet
+            self._create_empty_file(local_file_name_abs_path)
+        elif not self.os_service.exists(local_file_name_abs_path) and len(artifact.files) > 0:
+            self._copy_remote_file(artifact, local_file_name_abs_path)
+
+    def _direct_to_cache(self, local_file_name_abs_path, cache_directory, cache_path):
+        if self.os_service.exists(cache_path):
+            self.logger.info("Fetching cached artifact from '{}'".format(cache_path))
+            self.os_service.copy(cache_path, ".")
+        elif self.os_service.exists(local_file_name_abs_path):
+            if not self.os_service.exists(cache_directory):
+                self.os_service.mkdir(cache_directory)
+            self.logger.info("Copying artifact to cache directory, {}=>{}".format(
+                local_file_name_abs_path, cache_directory))
+            self.os_service.copy(local_file_name_abs_path, cache_directory)
+
+    def _create_empty_file(self, local_file_name_abs_path):
+        with self.os_service.open_file(local_file_name_abs_path, "w+"):
```

Comment:
This variable name could perhaps be called "file_path" or similar, as the function doesn't require an absolute path.